src/libraries/System.Linq/src/System/Linq/Min.cs

```diff
@@ -132,151 +136,58 @@ private static int Min(ReadOnlySpan<int> span)
-                    {
-                        value = x;
-                    }
-                }
-            }
+        public static double Min(this IEnumerable<double> source) => MinFloat(source);
 
-            return value;
-        }
+        public static double? Min(this IEnumerable<double?> source) => MinFloat(source);
 
-        private static long Min(ReadOnlySpan<long> span)
+        private static T MinFloat<T>(this IEnumerable<T> source) where T : struct, IFloatingPointIeee754<T>
         {
-            if (span.IsEmpty)
-            {
-                ThrowHelper.ThrowNoElementsException();
-            }
+            T value;
 
-            // Vectorize the search if possible.
-            int index;
-            long value;
-            if (Vector.IsHardwareAccelerated && span.Length >= Vector<long>.Count * 2)
+            if (source.TryGetSpan(out ReadOnlySpan<T> span))
             {
-                // The span is at least two vectors long. Create a vector from the first N elements,
-                // and then repeatedly compare that against the next vector from the span.  At the end,
-                // the resulting vector will contain the minimum values found, and we then need only
-                // to find the min of those.
-                var mins = new Vector<long>(span);
-                index = Vector<long>.Count;
-                do
+                if (span.IsEmpty)
                 {
-                    mins = Vector.Min(mins, new Vector<long>(span.Slice(index)));
-                    index += Vector<long>.Count;
+                    ThrowHelper.ThrowNoElementsException();
                 }
-                while (index + Vector<long>.Count <= span.Length);
 
-                value = mins[0];
-                for (int i = 1; i < Vector<long>.Count; i++)
-                {
-                    if (mins[i] < value)
-                    {
-                        value = mins[i];
-                    }
-                }
-            }
-            else
-            {
                 value = span[0];
-                index = 1;
-            }
-
-            // Iterate through the remaining elements, comparing against the min.
-            for (int i = index; (uint)i < (uint)span.Length; i++)
-            {
-                if (span[i] < value)
+                for (int i = 1; (uint)i < (uint)span.Length; i++)
                 {
-                    value = span[i];
-                }
-            }
-
-            return value;
-        }
-
-        public static long? Min(this IEnumerable<long?> source)
-        {
-            if (source == null)
-            {
-                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.source);
-            }
-
-            long? value = null;
-            using (IEnumerator<long?> e = source.GetEnumerator())
-            {
-                do
-                {
-                    if (!e.MoveNext())
+                    T f = span[i];
```

Comment:
nit: `f` isn't a great name

Reply:
Yeah, this was there before, but it's even more strange now that it's no longer limited to "f"loats.  Changed.